packages/grpc-js-core/src/channel.ts

```diff
@@ -3,16 +3,26 @@ import * as http2 from 'http2';
 import {checkServerIdentity, SecureContext, PeerCertificate} from 'tls';
 import * as url from 'url';
 
+import {Call} from './call';
 import {CallCredentials} from './call-credentials';
 import {CallCredentialsFilterFactory} from './call-credentials-filter';
-import {CallOptions, CallStream, CallStreamOptions, Http2CallStream} from './call-stream';
+import {CallStream, CallStreamOptions, Http2CallStream} from './call-stream';
 import {ChannelCredentials} from './channel-credentials';
 import {CompressionFilterFactory} from './compression-filter';
+import {EmitterAugmentation0} from './events';
 import {Status} from './constants';
 import {DeadlineFilterFactory} from './deadline-filter';
 import {FilterStackFactory} from './filter-stack';
 import {Metadata, MetadataObject} from './metadata';
 import { MetadataStatusFilterFactory } from './metadata-status-filter';
+import { PropagateFlags } from './index';
+
+export type CallOptions = {
+  // Represents a parent server call.
+  // For our purposes we only need to know of the 'cancelled' event.
+  parent?: EmitterAugmentation0<'cancelled'> & EventEmitter;
```

Comment:
We have to be careful doing something like this, and I think it will have to change before the actual 1.0 release. The problem is that if we further specify this type, for propagating other things, we can break users who were obeying the previous type constraint.

Reply:
I wasn't planning on making any additional assumptions about what the server-side call object looks like. Given that this is how cancellation is propagated in `grpc` currently (with a 'cancelled' event), I thought this would be a safe and minimal assumption to make.

I can drop this commit entirely and skip tests that rely on it for now -- but I don't think it's necessary assuming that we will eventually have a document detailing the exact interactions between client and server before the 1.0 release anyway. What do you think?